### PR TITLE
build: 2.5.0

### PR DESCRIPTION
**Pull request by Pakisan**
### Added

- IDEA 2024.1

### Fixed

- Incorrect validation of Multi Format Schema and bindings

### Changed
- 2.0.0 schema validation and completion:
  - Updated allowed extension name: `^x-[\w\d\-\_]+$` was changed to `^x-[\w\d\.\x2d_]+$`
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 2.1.0 schema validation and completion:
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 2.2.0 schema validation and completion:
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 2.3.0 schema validation and completion:
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 2.4.0 schema validation and completion:
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 2.5.0 schema validation and completion:
  - OpenAPI schema v3
  - Added `schemaFormat` to messages to recognize properly non AsyncAPI Schemas
  - Was updated `JsonSchema` Draft 07
- 3.0.0 schema validation and completion:
  - Bindings:
    - Solace:
      - server binding 0.4.0
      - operation binding 0.4.0
    - HTTP:
      - message binding 0.3.0
      - operation binding 0.3.0

### CODE REVIEW

**Comment by theschles** [comment]
Hi @Pakisan just saw this.  Will look at it when I get into the office later today.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (philCryoport) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why not gradle 8.7?
https://github.com/gradle/gradle/releases

**Review by philCryoport — COMMENTED** [review]
How did you verify your JSON changes didn't break anything?

**Comment by philCryoport** [comment]
Decided to RTFM `./gradlew`

Ran `./gradlew check`

<details>
<summary>Screen output from `./gradlew check` </summary>

```
jasyncapi-idea-plugin on  release/2.5.0 is 📦 2.5.0+jre17 via ☕ v21.0.2 via 🅒 base
➜ ./gradlew :check

> Task :verifyPluginConfiguration
[gradle-intellij-plugin :verifyPluginConfiguration] The following plugin configuration issues were found:
- The 'since-build' property is lower than the target IntelliJ Platform major version: 223 < 241.
- The dependency on the Kotlin Standard Library (stdlib) is automatically added when using the Gradle Kotlin plugin and may conflict with the version provided with the IntelliJ Platform, see: https://jb.gg/intellij-platform-kotlin-stdlib
See: https://jb.gg/intellij-platform-versions

> Task :test
CompileCommand: exclude com/intellij/openapi/vfs/impl/FilePartNodeRoot.trieDescend bool exclude = true

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:162

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:162

JsonReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at JsonReferenceVariantsTest.kt:161

YamlReferenceVariantsTest > test referenced file FAILED
    junit.framework.AssertionFailedError at YamlReferenceVariantsTest.kt:160

431 tests completed, 16 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///foo/asyncapi/jasyncapi-idea-plugin/build/reports/tests/test/index.html

* Try:
> Run with --scan to get full insights.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 31s
14 actionable tasks: 8 executed, 6 up-to-date
```

</details>

Here's the full output
[test.zip](https://github.com/asyncapi/jasyncapi-idea-plugin/files/15046249/test.zip)

**Comment by Pakisan** [comment]
> How did you verify your JSON changes didn't break anything?

Now:
- I have repo with bunch of different specifications, where I test bundled JsonSchema for correctness
- Tests inside project to validate correctness of auto-completion which is relaying on JsonSchema

In future:
On May, I hope, I'll reanimate our TCK repo to switch to it for schema tests

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Pakisan) [reply gradle/wrapper/gradle-wrapper.properties:3]: Will switch to it on next release. Honestly, idk why I have ignored 8.7

Anyway it affects nothing, except features like increment compiling and so on

**Comment by philCryoport** [comment]
> > For record's sake, I ran `:runPluginVerifier`. Result:
> > ```
> > BUILD SUCCESSFUL in 11m 3s
> > 17 actionable tasks: 8 executed, 9 up-to-date
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Ahem...then decided to RTFM `./gradlew`...
> > Screen output from `./gradlew check`
> > Here's the full output [test.zip](https://github.com/asyncapi/jasyncapi-idea-plugin/files/15046249/test.zip)
> 
> This is fine😓
> 
> This is so old, that I'm resigned with it. Those tests are checking single scenario - `$ref` to external file.
> 
> At runtime everything is working great, but in test context it fails.
> 
> I think that will fix it in parallel with #47, cause it's about digging into JetBrains sdk to fix it

Issue created https://github.com/asyncapi/jasyncapi-idea-plugin/issues/59

**Comment by philCryoport** [comment]
> > How did you verify your JSON changes didn't break anything?
> 
> Now:
> 
>     * I have repo with bunch of different specifications, where I test bundled JsonSchema for correctness
> 
>     * Tests inside project to validate correctness of auto-completion which is relaying on JsonSchema
> 
> 
> In future: On May, I hope, I'll reanimate our TCK repo to switch to it for schema tests

Issue created https://github.com/asyncapi/jasyncapi-idea-plugin/issues/60

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (philCryoport) [reply gradle/wrapper/gradle-wrapper.properties:3]: Also patches a High-severity CVE: https://github.com/asyncapi/jasyncapi-idea-plugin/issues/61

**Review by philCryoport — APPROVED** [review]
Noted various improvements in issues.

**Comment by theschles** [comment]
@theschles is my personal Github; @philCryoport is my work account.

**Pull request merged**